src/sst/core/elemLoader.cc

```diff
@@ -135,92 +73,114 @@ splitPath(const std::string& searchPaths)
+    const char* verbose_env = getenv("SST_CORE_DL_VERBOSE");
+    if ( nullptr != verbose_env ) { verbose = atoi(verbose_env) > 0; }
 
-        fullpath = path + "/" + so_path;
-        ret      = stat(fullpath.c_str(), &sbuf);
-        if ( ret == 0 ) break;
+    const char* bind_env = getenv("SST_CORE_DL_BIND_POLICY");
+    if ( nullptr == bind_env ) { bindPolicy = RTLD_LAZY | RTLD_GLOBAL; }
+    else if ( (!strcmp(bind_env, "lazy")) || (!strcmp(bind_env, "LAZY")) ) {
+        bindPolicy = RTLD_LAZY | RTLD_GLOBAL;
     }
-
-    // This is a little weird, but always try the last path - if we
-    // didn't succeed in the stat, we'll get a file not found error
-    // from dlopen, which is a useful error message for the user.
-    handle = dlopen(fullpath.c_str(), RTLD_NOW | RTLD_GLOBAL);
-    if ( nullptr == handle ) {
-        std::vector<char> err_str(1e6); // make darn sure we fit the str
-        sprintf(
-            err_str.data(),
-            "Opening and resolving references for element library %s failed:\n"
-            "\t%s\n",
-            elemlib.c_str(), dlerror());
-        err_os << (const char*)err_str.data();
+    else if ( (!strcmp(bind_env, "now")) || (!strcmp(bind_env, "NOW")) ) {
+        bindPolicy = RTLD_NOW | RTLD_GLOBAL;
     }
 }
 
+ElemLoader::~ElemLoader() {}
+
 void
 ElemLoader::loadLibrary(const std::string& elemlib, std::ostream& err_os)
 {
-    std::string libname = "lib" + elemlib;
-    lt_dlhandle lt_handle;
-    lt_handle = lt_dlopenadvise(libname.c_str(), loaderData->advise_handle);
-    if ( nullptr == lt_handle ) {
-        // The preopen module runs last and if the
-        // component was found earlier, but has a missing symbol or
-        // the like, we just get an amorphous "file not found" error,
-        // which is totally useless...
-        // Make darn sure we have enough space to hold the error message
-        std::vector<char> err_str(1e6);
-        sprintf(err_str.data(), "Opening element library %s failed: %s\n", elemlib.c_str(), lt_dlerror());
-        err_os << (const char*)err_str.data();
-        followError(libname, elemlib, searchPaths, err_os);
-    }
+    std::vector<std::string> paths   = splitPath(searchPaths);
+
+    char* full_path     = new char[PATH_MAX];
+    bool  found_element = false;
 
-    // loading a library can "wipe" previously loaded libraries depending
-    // on how weak symbol resolution works in dlopen
-    // rerun the loaders to make sure everything is still registered
-    for ( auto& libpair : ELI::LoadedLibraries::getLoaders() ) {
-        // loop all the elements in the element lib
-        for ( auto& elempair : libpair.second ) {
-            // loop all the loaders in the element
-            for ( auto* loader : elempair.second ) {
-                loader->load();
+    for ( std::string& next_path : paths ) {
+        if ( verbose ) { printf("SST-DL: Searching: %s\n", next_path.c_str()); }
+
+        if ( next_path.at(next_path.size() - 1) == '/' ) {
```

Comment:
`std::string` has a `.back()` method.

Reply:
Also `.at()`  does bounds checking, but that is pessimistic here since `size() - 1` should be fine.

Reply:
This one needs to happen